pytorch_lightning/trainer/trainer.py

```diff
@@ -2102,13 +2106,13 @@ def _results(self) -> Optional[ResultCollection]:
     def _exit_gracefully_on_signal(self) -> None:
         if not _fault_tolerant_training():
             return
-        if not self._should_terminated_gracefully():
+        if not self._should_terminate_gracefully():
```

Comment:
```suggestion
        if not _fault_tolerant_training() or not self._should_terminate_gracefully():
```

Reply:
> No, you don't want to force a reduce when fault tolerant is not active. This would be costly for the users who don't want to opt-in.

If fault tolerant training is not active, `not _fault_tolerant_training() == True` and the right hand side is not evaluated

Reply:
I shared it before, we don't want to merge this. This would impact performance for user which don't opt-in on fault tolerant training.

Reply:
See my last comment

```python
def _fault_tolerant_training():
    return False


def _should_terminate_gracefully():
    assert False


def foo():
    if not _fault_tolerant_training() or not _should_terminate_gracefully():
        print("Just return")
        return
    print("Doing stuff")

foo()
```

Reply:
Done !